nix/module/server.nix

```diff
@@ -0,0 +1,111 @@
+{ pkgs, lib, config, ... }:
+
+{
+  options.services.filetracker = {
+    enable = lib.mkEnableOption "filetracker server";
+
+    package = lib.mkPackageOption pkgs "Filetracker" {
+      default = [ "python38Packages" "filetracker" ];
+    };
+
+    listenAddress = lib.mkOption {
+      default = "0.0.0.0";
+      description = "The address that filetracker-server will listen on";
+      type = lib.types.str;
+    };
+
+    port = lib.mkOption {
+      default = 9999;
+      description = "The port that filetracker-server will listen on";
+      type = lib.types.port;
+    };
+
+    openFirewall = lib.mkOption {
+      default = true;
+      description = "Whether to open the provided port in the firewall";
+      type = lib.types.bool;
+    };
+
+    ensureFiles = lib.mkOption {
+      default = { };
+      description = "Files that should be added to filetracker after start";
+      type = lib.types.attrsOf lib.types.path;
+    };
+
+    workers = lib.mkOption {
+      default = "auto";
+      description = "The number of gunicorn workers to spawn";
+      type = with lib.types; oneOf [ (strMatching "auto") ints.positive ];
+    };
+  };
+
+  config =
+    let
+      cfg = config.services.filetracker;
+      python = cfg.package.pythonModule;
+
+      createEnsureService = remotePath: localPath:
+        let
+          systemdEscapedPath = builtins.replaceStrings [ "/" ] [ "-" ] (lib.removePrefix "/" remotePath);
+          serviceName = "filetracker-put-${systemdEscapedPath}";
+        in
+        lib.nameValuePair serviceName {
+          enable = true;
+          description = "Filetracker ensure ${remotePath}";
+          after = [ "filetracker.service" ];
+          # We want to "stop" if ft is stopped, so we will run again when ft starts.
+          partOf = [ "filetracker.service" ];
+          # We need to start on creation, hence the "multi-user.target".
+          wantedBy = [ "multi-user.target" "filetracker.service" ];
```

Comment:
Why is `filetracker.service` not enough by itself?

Reply:
Don't really know, but new units didn't start by themselves upon creation.
Though it was before making them oneshot, so I will check if that is still the case.

Reply:
I checked and the comment is still right; we need the second target.